### PR TITLE
[DEVHUB-1744][DEVHUB-1745] Incorporate CS Podcasts and Videos

### PR DESCRIPTION
**Pull request by jackyw2018**
## Updates:

- [DEVHUB-1744: Integerate Podcasts](https://jira.mongodb.org/browse/DEVHUB-1744)
- [DEVHUB-1745: Integrate Videos](https://jira.mongodb.org/browse/DEVHUB-1745)

### CODE REVIEW

File: src/api-requests/contentstack_utils.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { UnderlyingClient } from '../types/client-factory';
+import { DocumentNode } from 'graphql';
+
+type gqlParents =
+    | 'l1Products'
```
Comment (jackyw2018) [inline src/api-requests/contentstack_utils.ts:5]: This is used in `fetchAll()` and may need to expand for missing parents, which matches to the top-level key in a gql:
```
videos: all_videos(where: { slug: $slug }, skip: $skip) {
    total
    items {
        ...VideoFields
    }
}
```

File: src/graphql/podcasts.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            }
+        }
+    }
+    ${podcastFields}
+`;
+
+export const podcastsBySlugQuery = gql`
+    query Podcasts($slug: String!, $skip: Int = 0) {
+        podcasts: all_podcasts(where: { slug: $slug }, skip: $skip) {
+            total
```
Comment (jackyw2018) [inline src/graphql/podcasts.ts:127]: `fetchAll()` works with both `allQuery` and `BySlugQuery`; `fetchAll()` assumes the `total` field exists in the query to work correctly;

I mentioned this in fetchAll()'s docstring but want to highlight this here.

File: src/utils/client-factory.ts [hunk 3]
```diff
@@ -52,8 +55,35 @@ const clientFactory = <T extends ClientType>(
...
+                                return fetch(...pl);
+                            }
+
+                            // https://github.com/apollographql/apollo-client/issues/4017
+                            // tweaked from musemind implementation
+
+                            // Uncomment below if we want to see outbound gql uri
+                            // const [uri] = pl;
+                            // console.log(uri)
+
```
Comment (jackyw2018) [inline src/utils/client-factory.ts:81]: I commented out this part, but still wanted to kept this in case we need to log the URI for data mocking purposes

**Review by mcolella14 — COMMENTED** [review]
Looks good, but you will have to add a handler for the mocks here: https://github.com/mongodb/devcenter/blob/cms_migration/mocks/handlers.ts#L81

**Review by harikakotipalli — APPROVED** [review]

**Pull request merged**